emop-core/src/main/java/edu/cornell/emop/util/Util.java

```diff
@@ -47,4 +50,54 @@ public static void replaceFileInJar(String jarPath, String oldPath, String newPa
             ex.printStackTrace();
         }
     }
+
+    /**
+     * Recursive routine accumulating the set of package names within the project.
+     * Borrowed from DSI.
+     *
+     * @param currRoot the current location
+     * @param dirName the name of the directory
+     * @return set of all package names within the project.
+     */
+    private static Set<String> classFilesWalk(File currRoot, String dirName) {
+        Set<String> packageNameSet = new HashSet<>();
+        File[] files = currRoot.listFiles();
+        if (files == null) {
+            return packageNameSet;
+        }
+        // we want to list all the potential *.class files in this directory
+        File[] classFiles = currRoot.listFiles(new FileFilter() {
+            @Override
+            public boolean accept(File file) {
+                return !file.isDirectory() && (file.getName().endsWith(".class"));
+            }
+        });
+        if (classFiles.length > 0) {
+            // we found a class file, which means we are in a directory with class files (package)
+            // No need to traverse further because AspectJ within(${PACKAGE_NAME}..*) syntax instruments subpackages.
+            String packageName = currRoot.getAbsolutePath().split(dirName + File.separator)[1];
```

Comment:
what assumptions are you making about what the path looks like here? Can you give an example and add in the comment what the `[1]` index points to?

Reply:
Do you want me to add this into the comments above line 78?

Reply:
Please look through the updated comments and JavaDoc for the method and let me know if they effectively convey the assumptions!